src/data/sym2.lean

```diff
@@ -0,0 +1,263 @@
+/-
+Copyright (c) 2020 Kyle Miller All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Kyle Miller.
+-/
+
+import data.quot
+import tactic
+open function
+
+/-!
+# The symmetric square
+
+This file defines symmetric squares, which is `α × α` modulo swapping.
+This is also known as the type of unordered pairs.
+
+From the point of view that an unordered pair is equivalent to a
+multiset of cardinality two, we have `sym2.mem` (the `has_mem`
```

Comment:
I would prefer having an `equiv` with `{s : multiset X // s.card = 2}`, and I suggest explicitly naming this result in the docstring for `sym2`, to make the connection with the multiset API.

Reply:
Defined it (in the newest commit).  It seemed easiest to define symmetric powers in general due to some issues I was having getting a `quot.rec` to work out properly...  Now that I got it all to work, I could probably extract a more direct proof, if you think that would be better.

Reply:
@kmill Thanks a lot, I think it's helpful. However, now the definition of `sym` looks a bit out of place. I suggest that you start a new file for this `data/sym` and put the definition (also `sym'`) and basic lemmas in there.
I won't ask you to flesh out the API of `sym` any further, just put a TODO/warning at the top of the file, that this is only there as supporting material for `sym2`.
Another PR could add a lot more stuff to the `sym` file.

Reply:
@jcommelin Ok, done!